### PR TITLE
TeacherAttachment Controller Get by Id

### PR DESCRIPTION
**Pull request by viralpandya**

### CODE REVIEW

File: OtripleS.Web.Api/Services/TeacherAttachments/TeacherAttachmentService.cs [hunk 1]
```diff
@@ -36,5 +36,20 @@ public partial class TeacherAttachmentService : ITeacherAttachmentService
 
             return await this.storageBroker.InsertTeacherAttachmentAsync(teacherAttachment);
         });
+
+        public ValueTask<TeacherAttachment> RetrieveTeacherAttachmentByIdAsync(
+            Guid teacherId,
+            Guid attachmentId) =>
+        TryCatch(async () =>
```
Comment (hassanhabib) [inline OtripleS.Web.Api/Services/TeacherAttachments/TeacherAttachmentService.cs:43]: Move this up by one

File: OtripleS.Web.Api/Services/TeacherAttachments/TeacherAttachmentService.Validations.cs [hunk 2]
```diff
@@ -41,5 +41,16 @@ private void ValidateTeacherAttachmentIdIsNull(Guid teacherId, Guid attachmentId
                     parameterValue: attachmentId);
             }
         }
+
+        private static void ValidateStorageTeacherAttachment(
+            TeacherAttachment storageTeacherAttachment,
+            Guid studentId, 
+            Guid attachmentId)
+        {
+            if (storageTeacherAttachment == null)
```
Comment (hassanhabib) [inline OtripleS.Web.Api/Services/TeacherAttachments/TeacherAttachmentService.Validations.cs:50]: Do `is null`

File: OtripleS.Web.Api/Services/TeacherAttachments/TeacherAttachmentService.cs (hunk 1 above)
Comment (viralpandya) [reply OtripleS.Web.Api/Services/TeacherAttachments/TeacherAttachmentService.cs:42]: Resolved

File: OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                new TeacherAttachmentDependencyException(databaseUpdateException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectTeacherAttachmentByIdAsync(someTeacherId, someAttachmentId))
+                    .ThrowsAsync(databaseUpdateException);
+
+            // when
+            ValueTask<TeacherAttachment> retrieveTeacherAttachmentTask =
+                this.teacherAttachmentService.RetrieveTeacherAttachmentByIdAsync
+                (someTeacherId, someAttachmentId);
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs:78]: Break this down

File: OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+
+            // then
+            await Assert.ThrowsAsync<TeacherAttachmentDependencyException>(
+                () => retrieveTeacherAttachmentTask.AsTask());
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs:82]: Move this guy up

File: OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+        {
+            // given
+            Guid randomAttachmentId = Guid.NewGuid();
+            Guid someAttachmentId = randomAttachmentId;
+            Guid randomTeacherId = Guid.NewGuid();
+            Guid someTeacherId = randomTeacherId;
+            SqlException sqlException = GetSqlException();
+
+            var expectedTeacherAttachmentDependencyException
+                = new TeacherAttachmentDependencyException(sqlException);
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs:30]: Move the `=` sign up

File: OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+
+            this.storageBrokerMock.Setup(broker =>
+                 broker.SelectTeacherAttachmentByIdAsync(someTeacherId, someAttachmentId))
+                    .ThrowsAsync(sqlException);
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs:34]: Fix tabs

File: OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+
+            // when
+            ValueTask<TeacherAttachment> retrieveTeacherAttachmentTask =
+                this.teacherAttachmentService.RetrieveTeacherAttachmentByIdAsync
+                (someTeacherId, someAttachmentId);
```
Comment (hassanhabib) [inline OtripleS.Web.Api.Tests.Unit/Services/TeacherAttachments/TeacherAttachmentServiceTests.Exceptions.RetrievById.cs:39]: Break input parameters down

**Pull request merged**